File: dcp-solutions/2024-12-07/solution.py
```python
def divide(dividend: int, divisor: int) -> int:
    # Define signs of result
    sign = -1 if ((dividend < 0) ^ (divisor < 0)) else 1
    
    # Convert to positive
    dividend, divisor = abs(dividend), abs(divisor)
    
    quotient = 0
    while dividend >= divisor:
        temp, i = divisor, 1
        while dividend >= (temp << 1):
            temp <<= 1
            i <<= 1
        dividend -= temp
        quotient += i
    
    return sign * quotient
```

File: dcp-solutions/2024-12-07/solution.php
```php
function divide($dividend, $divisor) {
    // Determine the sign of the result
    $sign = ($dividend < 0) !== ($divisor < 0);

    // Convert both numbers to positive
    $dividend = abs($dividend);
    $divisor = abs($divisor);

    // Calculate the quotient using repeated subtraction
    $quotient = 0;
    while ($dividend >= $divisor) {
        $dividend -= $divisor;
        $quotient++;
    }

    return $sign ? -$quotient : $quotient;
}
```